src/Component/CodeEditor/CodeEditor.tsx

```diff
@@ -237,20 +244,56 @@ export class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState
       style
     } = this.props;
     if (style) {
+      let fileName = style.name;
       let type = 'application/json;charset=utf-8';
       if (activeParser && activeParser.title === 'SLD Style Parser') {
         type = 'text/xml;charset=utf-8';
+        fileName += '.sld';
```

Comment:
Suggestion:

```suggestion
        if (!fileName.endsWith('.sld')) {
            fileName += '.sld';
        }
```

Reply:
As the filename is just `style.name` it is very unlikely that it ends withs '.sld'...